include/maths/common/CSolvers.h

```diff
@@ -570,7 +583,7 @@ class MATHS_COMMON_EXPORT CSolvers {
                 return true;
             }
 
-            if (fa * fs > 0.0) {
+            if (std::signbit(fa) == std::signbit(fs)) {
```

Comment:
Should this also confirm both variables are not equal to zero?  The new condition is slightly different in the case of one or both variables being zero with the same sign.